docs/guides/tutorials/nextjs.rst

```diff
@@ -0,0 +1,270 @@
+  ....... };
+  {default::BlogPost {id: 7f301d02-c780-11ec-8a1a-a34776e884a0}}
+  edgedb> insert BlogPost {
+  .......   title := "How to build a blog with EdgeDB and Next.js",
+  .......   content := "Let's start by scaffolding our app..."
+  ....... };
+  {default::BlogPost {id: 88c800e6-c780-11ec-8a1a-b3a3020189dd}}
+
+
+Alternatively, you can use the EdgeDB UI to manage your schema and data. Open the EdgeDB UI by running:
+
+.. code-block:: bash
+
+  $ npx edgedb ui
+
+
+Generating the query builder​ 
+-----------------------------
+
+This step is optional but recommended for TypeScript projects. The query builder generates TypeScript types for your EdgeQL queries:
+
+.. code-block:: bash
+
+  $ npx @edgedb/generate edgeql-js
+
+The command introspects your schema and generates a query builder in the ``dbschema/edgeql-js`` directory.
+
+Using EdgeDB in Next.js
+-----------------------
+
+EdgeDB with React Server Components (App Router)
+================================================
+
+Create a server component that fetches data directly from EdgeDB:
+
+.. code-block:: tsx
+  :caption: app/page.tsx
+
+  import { createClient } from 'edgedb';
+  import e from '@/dbschema/edgeql-js';
+
+  const client = createClient();
+
+  export default async function Posts() {
+    const posts = await e.select(e.BlogPost, () => ({
+      id: true,
+      title: true,
+      content: true,
+    })).run(client);
+
+    return (
+      <div>
+        {posts.map(post => (
+          <article key={post.id}>
+            <h2>{post.title}</h2>
+            <p>{post.content}</p>
+          </article>
+        ))}
+      </div>
+    );
+  }
+
+With API Routes
+===============
+
+Create an API route and fetch data from the client side:
+
+.. code-block:: tsx
+  :caption: pages/api/posts.ts
+
+  import type { NextApiRequest, NextApiResponse } from 'next';
+  import { createClient } from 'edgedb';
+  import e from '@/dbschema/edgeql-js';
+
+  const client = createClient();
+
+  export default async function handler(
+    req: NextApiRequest,
+    res: NextApiResponse
+  ) {
+    const posts = await e.select(e.BlogPost, () => ({
+      id: true,
+      title: true,
+      content: true,
+    })).run(client);
+    
+    res.status(200).json(posts);
+  }
+
+.. code-block:: tsx
+  :caption: pages/api/posts.ts
+
+  import { useEffect, useState } from 'react';
+
+  export default function Posts() {
+    const [posts, setPosts] = useState(null);
+
+    useEffect(() => {
+      fetch('/api/posts')
+        .then(res => res.json())
+        .then(setPosts);
+    }, []);
+
+    if (!posts) return <div>Loading...</div>;
+
+    return (
+      <div>
+        {posts.map(post => (
+          <article key={post.id}>
+            <h2>{post.title}</h2>
+            <p>{post.content}</p>
+          </article>
+        ))}
+      </div>
+    );
+  }
+
+Deployment
+-----------
+
+First, add a ``prebuild`` script to your ``package.json`` to generate the query builder during deployment:
```

Comment:
We should move this to `gel.toml` hooks

Reply:
But don't we need this, too? So that t works with a cloud provider when building the app. As I understand, the hooks will work after applying the migration etc, so if I deploy to e.g. Vercel and connect to the Cloud with env variables, it won't run the generator.

Reply:
yeah, good point, we should do both. We probably need some additional direction to get it to even _work_ during deployment since it needs access to a live database. I think it works fine in the Vercel integration which I'm pretty sure sets the `EDGEDB_INSTANCE` and `EDGEDB_SECRET_KEY` environment variables correctly.

Maybe this section should be merged into the sections below?